Patch-Source: https://github.com/extrawurst/gitui/pull/1323
--
From 82426f48569a1835636837f70954934cd9f0c16e Mon Sep 17 00:00:00 2001
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sat, 10 Sep 2022 19:04:36 +0200
Subject: [PATCH 2/2] allow to build without vendored openssl

Vendoring (bundling) openssl library is very bad for security and
Linux distributions forbid it. The aim of this change is to simplify
packaging gitui in linux distros.

diff --git a/Cargo.toml b/Cargo.toml
index 72537dc4b..b54884d0c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -65,13 +65,14 @@ pretty_assertions = "1.3"
 maintenance = { status = "actively-developed" }
 
 [features]
-default =["ghemoji", "regex-fancy", "trace-libgit"]
+default =["ghemoji", "regex-fancy", "trace-libgit", "vendor-openssl"]
 ghemoji =["gh-emoji"]
 # regex-* features are mutually exclusive.
 regex-fancy = ["syntect/regex-fancy"]
 regex-onig = ["syntect/regex-onig"]
 timing =["scopetime/enabled"]
 trace-libgit =["asyncgit/trace-libgit"]
+vendor-openssl = ["asyncgit/vendor-openssl"]
 
 [workspace]
 members =[
diff --git a/asyncgit/Cargo.toml b/asyncgit/Cargo.toml
index d430d5bb7..85a4aec1a 100644
--- a/asyncgit/Cargo.toml
+++ b/asyncgit/Cargo.toml
@@ -19,7 +19,7 @@ log = "0.4"
 # git2 = { path = "../../extern/git2-rs", features = ["vendored-openssl"]}
 # git2 = { git="https://github.com/extrawurst/git2-rs.git", rev="fc13dcc", features = ["vendored-openssl"]}
 # pinning to vendored openssl, using the git2 feature this gets lost with new resolver
-openssl-sys = { version = '0.9', features = ["vendored"] }
+openssl-sys = { version = '0.9', features = ["vendored"], optional = true }
 rayon-core = "1.9"
 scopetime = { path = "../scopetime", version = "0.1" }
 shellexpand = "2.1"
@@ -37,3 +37,4 @@ tempfile = "3.2"
 [features]
 default = ["trace-libgit"]
 trace-libgit = []
+vendor-openssl = ["openssl-sys"]
